fedimint-core/src/module/registry.rs

```diff
@@ -2,46 +2,50 @@ use std::collections::BTreeMap;
 
 use anyhow::anyhow;
 
-use crate::core::Decoder;
 pub use crate::core::ModuleInstanceId;
+use crate::core::{Decoder, ModuleKind};
 use crate::server::DynServerModule;
 
 /// Module Registry hold module-specific data `M` by the `ModuleInstanceId`
 #[derive(Debug, Clone)]
-pub struct ModuleRegistry<M>(BTreeMap<ModuleInstanceId, M>);
+pub struct ModuleRegistry<M>(BTreeMap<ModuleInstanceId, (ModuleKind, M)>);
```

Comment:
Do we really need it? Seems a bit unnecessary. For some types, you can just ask the `M` what's the `ModuleKind`, for some others it shouldn't matter?

Reply:
I don't know what's the exact reason for it. Is it for `ModuleDecoderRegistry`? We could add `fn kind(&self) -> ModuleKind` to `Decoder` instead?

Reply:
The reason was that I wanted to find modules by kind, we could also just extend the traits  to give us the module kind, just seemed more involved.

Reply:
Opened https://github.com/fedimint/fedimint/issues/2319 to track

Reply:
> The reason was that I wanted to find modules by kind,

Also a method, using `filter` :)